editor.planx.uk/src/contexts/ToastContext.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import ToastContainer from "components/Toast/ToastContainer";
+import {
+  ToastContextType,
+  ToastState,
+  ToastType,
+} from "components/Toast/types";
+import React, { createContext, ReactNode, useReducer } from "react";
+import { toastReducer } from "reducers/toastReducer";
+
+const defaultCreateContextValue = {
+  remove: (_id: number) => {},
+  addToast: (_type: ToastType, _message: string) => {},
+  success: (_message: string) => {},
+  warning: (_message: string) => {},
+  info: (_message: string) => {},
+  error: (_message: string) => {},
+};
+
+export const ToastContext = createContext<ToastContextType>(
+  defaultCreateContextValue,
+);
+
+const initialState: ToastState = {
+  toasts: [],
+};
+
+export const ToastContextProvider = ({
+  children,
+}: Readonly<{ children: ReactNode }>) => {
+  const [state, dispatch] = useReducer(toastReducer, initialState);
+  const addToast = (type: ToastType, message: string) => {
+    const id = Math.floor(Math.random() * 10_000_000);
```

Comment:
nit: we also have the `uuid` package available on the frontend which would let you generate a unique uuid id like `const id = uuidV4()`(consistent with how we generate session ids) - happy for this to work either way though / chances of unintentionally colliding/non-unique IDs seems veryyy low here :slightly_smiling_face:

Reply:
Nice tip thanks! I'll include that in the next PR 👍